### ISSUE TITLE
No functions available with Graphite data source

### ISSUE THREAD

**Issue by Hrimnir** [grafana/grafana#46951]
In previous versions (e.g. 8.5.0-50432pre, 7th Feb 2022 (git sha eb9d85a2bf)) of Grafana in the explore view, clicking on the "+"-button on the "functions"-line would provide a selection of functions to use with a query.

In newer versions (e.g. 8.4.3, 2nd March 2022 (git sha b7d2911cac)) the function list is empty.

There is a parse error in fetch.ts stating that the string returned by the functions-request is invalid JSON - which is correct. The string contains the JavaScript 'Inifinite' as a number value, which is not valid JSON. The string seems to be a JavaScript string which would need to be "parsed" using eval() and not parsed as a JSON-string.

I tested once using the wrong IP for the Graphite back-end and the same call returned 502 Bad Gateway - but in this case the selection of functions was available - even though the request failed. I can only assume that Grafana has a pre-defined set of (Graphite) functions it uses when the request fails.

This has lead me to belive that the commit 1e21184f65c76aa37ef6288b71f9d37ada6029fc (Fix: JSON parsing in backend_srv (https://github.com/grafana/grafana/pull/45598)), which now makes Grafana properly handle the JSON parse error in newer versions and return an empty object instead of failing, results in the empty object being used as the selection of functions, i.e. no functions available, instead of the pre defined set of functions, which was used earlier when this call failed.

Given that my assumptions are not completely off, a quick-"fix" could be to use the built in set of functions when the call returns an empty object, not just when failing exceptionally? Even though this would not use the actual list of functions provided by the Graphite data source, this would be much better than nothing. Not sure how to solve the real issue of parsing the JavaScript-string provided by Graphite (unless one used eval()) - but I would be happy with the pre-defined set of functions available in Grafana.

Obviously I'm making many assumptions here, but hope this can be helpful in solving the issue.

**Comment by zuchka** [comment]
Thanks for creating this issue, @Hrimnir 

I tested this issue on `v8.5.0-pre (9fc9708ba5)` and I could not reproduce. The `+` button next to functions worked as expected.

Can you please test again and, if there is still an issue, provide more detailed steps for reproduction? Thank You!

**Comment by Hrimnir** [comment]
Thanks for reporting back, @zuchka!

I see I didn't do my homework properly! :-) I tested with the latest version in main and you're correct, it does work with the latest version. I also found that someone had already committed the fix when I wrote this issue and there where 2 other issues already mentioning this. I must have been sleep walking at the time I reported the issue.

Appologies for not properly investigating the issue on my side before submitting this report.

Regards
